hashes/src/internal_macros.rs

```diff
@@ -209,6 +209,14 @@ macro_rules! hash_type {
                 Self::from_engine(engine)
             }
 
+            /// Hashes the entire contents of the `reader`.
+            #[cfg(all(feature = "std", feature = "bitcoin-io"))]
+            pub fn hash_reader<R: std::io::Read>(reader: &mut R) -> Result<Self, std::io::Error> {
+                let mut engine = Self::engine();
+                std::io::copy(reader, &mut engine)?;
```

Comment:
I was wondering if this is too much but it's likely actually clever - it might give us unstable buffer optimizations on stable, so ACK this approach.

Reply:
I'll take it, but I just copied the rustdocs from `lib.rs` :)

Reply:
I mean if it's too much going through writer rather than doing `read`s into a buffer that gets hashed.

Reply:
Flagging that this thread may be unresolved. To me its resolved, with no action.